### PR TITLE
Enable quota-based instant delegation with era restrictions

### PR DESCRIPTION
**Pull request by ianhe8x**
---

### Description

This pull request introduces a quota-based instant delegation feature with configurable limits to enhance staking security and efficiency. Key features include:

- **Per-wallet, per-era quotas** for instant delegation, auto-resetting with each era transition.
- **70% era window restriction** to prevent last-minute sniping.
- Automatic **splitting of large delegations** into instant and pending parts.
- Reuse of existing `applyRedelegation()` for instant rewards processing.
- Configurable parameters for quota limits and era window percentage.

### Key Changes

- **Staking.sol**: Implements quota tracking, token transfers, and configuration options.
- **StakingManager.sol**: Refactored `delegate()` logic with era progress and quota handling.
- **IStaking.sol**: Introduces new interface methods supporting instant delegation.
- Comprehensive **test coverage** for various instant delegation scenarios.

### Checklist

- [ ] Tests added for all new functionalities
- [ ] Documentation updated as needed
- [ ] Code formatting and linting check passes
- [ ] All previous unit tests pass

## Summary by CodeRabbit

* **New Features**
  * Instant delegation: per-wallet, era-scoped quota with era-window-aware handling, transfer-plus-delegation flow, configurable parameters, and remaining-quota readouts.

* **Tests**
  * Comprehensive tests for instant vs pending delegation, era-window behavior, quota reset/isolation, edge cases, parameter validation, and boundaries.

* **Chores**
  * Tooling upgraded to Yarn 4; build/config tooling and verification plugin updated; dependency pins adjusted.

* **Bug Fixes**
  * Fixed async assertion in vesting test to properly await revert/success expectations.

### CODE REVIEW

**Comment by ianhe8x** [comment]
@coderabbitai

**Comment by ianhe8x** [comment]
@coderabbitai review the changes

File: contracts/StakingManager.sol [hunk 1]
```diff
@@ -50,18 +52,59 @@ contract StakingManager is IStakingManager, Initializable, OwnableUpgradeable {
...
+        uint256 eraProgress = _calculateEraProgress();
+        uint256 windowPercent = staking.instantEraWindowPercent();
+        bool inInstantWindow = windowPercent > 0 && eraProgress <= windowPercent;
+
+        if (!inInstantWindow) {
+            // After window: all delegation is pending
+            staking.addDelegation(msg.sender, _runner, _amount, false);
+            return;
+        }
+
```
[the remark this answers is not in the document]
Comment (ianhe8x) [reply contracts/StakingManager.sol:84]: solved. this is more a doc issue.
